### PR TITLE
[python] Add FUSE support for REST Catalog

### PR DESCRIPTION
**Pull request by shyjsarah**
### Purpose

Linked issue: close #xxx

Add FUSE (Filesystem in Userspace) support for PyPaimon REST Catalog. When remote object storage paths (OSS, S3, HDFS, etc.) are mounted locally via FUSE, users can now access data directly through local filesystem paths for better performance, bypassing remote storage SDKs.

Key features:
 - Three configuration options: fuse.local-path.enabled, fuse.local-path.root, fuse.local-path.validation-mode
 - Three validation modes: strict (throw exception), warn (fallback), none (skip validation)
 - Validation checks default database location to verify FUSE mount
 - Automatic path conversion: skip catalog/bucket level from remote path
 - Tri-state validation state to avoid repeated validation

### Tests

- pypaimon/tests/rest/test_fuse_local_path.py - 13 unit test cases covering:
       - Path conversion logic (_resolve_fuse_local_path)
       - Validation modes (strict/warn/none)
       - file_io_for_data behavior with FUSE enabled/disabled
       - Edge cases (missing root config, no location, etc.)

### API and Format

No API or storage format changes. This is a new feature that adds optional configuration.

### Documentation

- Added docs/content/pypaimon/fuse-support.md - User documentation covering configuration, validation modes, usage examples, and limitations.

### Generative AI tooling

 Generated-by: Qwen Code

### CODE REVIEW

File: docs/content/pypaimon/fuse-support.md [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+
+When using PyPaimon REST Catalog to access remote object storage (such as OSS, S3, or HDFS), data access typically goes through remote storage SDKs. However, in scenarios where remote storage paths are mounted locally via FUSE (Filesystem in Userspace), users can access data directly through local filesystem paths for better performance.
+
+This feature enables PyPaimon to use local file access when FUSE mount is available, bypassing remote storage SDKs.
+
+## Configuration
+
+| Option | Type | Default | Description |
+|--------|------|---------|-------------|
+| `fuse.local-path.enabled` | Boolean | `false` | Whether to enable FUSE local path mapping |
```
Comment (JingsongLi) [inline docs/content/pypaimon/fuse-support.md]: fuse is local?  Just remove `local-path`?
- fuse.enabled
- fuse.root
- fuse.validation-mode

File: paimon-python/pypaimon/catalog/rest/rest_catalog.py [hunk 2]
```diff
@@ -385,9 +401,125 @@ def file_io_from_options(self, table_path: str) -> FileIO:
         return FileIO.get(table_path, self.context.options)
 
     def file_io_for_data(self, table_path: str, identifier: Identifier):
+        """
+        Get FileIO for data access, supporting FUSE local path mapping.
+        """
+        # Try to use FUSE local path
+        if self.fuse_local_path_enabled:
```
Comment (JingsongLi) [inline paimon-python/pypaimon/catalog/rest/rest_catalog.py]: Can you extract fuse codes to another py file? And import it if `fuse_enabled` only.

File: docs/content/pypaimon/fuse-support.md (hunk 1 above)
Comment (shyjsarah) [reply docs/content/pypaimon/fuse-support.md]: updated

File: paimon-python/pypaimon/catalog/rest/rest_catalog.py (hunk 2 above)
Comment (shyjsarah) [reply paimon-python/pypaimon/catalog/rest/rest_catalog.py]: updated
